### PR TITLE
refactor: add type hints to requests.utils

### PR DESCRIPTION
**Pull request by kkirsche**
This adds missing type hints to `requests.utils` methods as seen in:
https://github.com/psf/requests/blob/main/requests/utils.py

The `_Uri` type is a `TypeAlias` that was determined to be `str | bytes` through the use of `urlparse` in the function/method implementations

### CODE REVIEW

File: stubs/requests/requests/utils.pyi [hunk 1]
```diff
@@ -1,54 +1,57 @@
-from collections.abc import Iterable, Mapping
-from typing import Any, AnyStr
+from collections.abc import Generator, Iterable, Mapping
+from typing import Any, AnyStr, FrozenSet, TypeAlias
```
Comment (hauntsaninja) [inline stubs/requests/requests/utils.pyi:2]:
```suggestion
from typing import Any, AnyStr, FrozenSet
from typing_extensions import TypeAlias
```

File: stubs/requests/requests/utils.pyi [hunk 2]
```diff
@@ -1,54 +1,58 @@
...
-def dotted_netmask(mask): ...
-def is_ipv4_address(string_ip): ...
-def is_valid_cidr(string_network): ...
-def set_environ(env_name, value): ...
-def should_bypass_proxies(url, no_proxy: Iterable[AnyStr] | None) -> bool: ...
-def get_environ_proxies(url, no_proxy: Iterable[AnyStr] | None = ...) -> dict[Any, Any]: ...
-def select_proxy(url: str, proxies: Mapping[Any, Any] | None): ...
-def default_user_agent(name=...): ...
-def default_headers(): ...
+def unquote_unreserved(uri: _Uri) -> str: ...
```
Comment (JelleZijlstra) [inline stubs/requests/requests/utils.pyi:39]: It does `uri.split("%")`, so this only works with str.
```suggestion
def unquote_unreserved(uri: str) -> str: ...
```

File: stubs/requests/requests/utils.pyi [hunk 3]
```diff
@@ -1,54 +1,58 @@
...
+def requote_uri(uri: _Uri) -> str: ...
```
Comment (JelleZijlstra) [inline stubs/requests/requests/utils.pyi:40]:
```suggestion
def requote_uri(uri: str) -> str: ...
```
Calls `unquote_unreserved(uri)`

File: stubs/requests/requests/utils.pyi [hunk 4]
```diff
@@ -1,54 +1,58 @@
...
+def address_in_network(ip: str, net: str) -> bool: ...
+def dotted_netmask(mask: int) -> str: ...
+def is_ipv4_address(string_ip: str) -> bool: ...
+def is_valid_cidr(string_network: str) -> bool: ...
+def set_environ(env_name: str, value: None) -> Generator[None, None, None]: ...
```
Comment (JelleZijlstra) [inline stubs/requests/requests/utils.pyi:45]:
```suggestion
def set_environ(env_name: str, value: None) -> ContextManager[None]: ...
```
It has `@contextmanager`. May need to import it from contextlib.

File: stubs/requests/requests/utils.pyi [hunk 5]
```diff
@@ -1,54 +1,58 @@
...
+def should_bypass_proxies(url: _Uri, no_proxy: Iterable[AnyStr] | None) -> bool: ...
+def get_environ_proxies(url: _Uri, no_proxy: Iterable[AnyStr] | None = ...) -> dict[Any, Any]: ...
```
Comment (JelleZijlstra) [inline stubs/requests/requests/utils.pyi:47]:
```suggestion
def should_bypass_proxies(url: _Uri, no_proxy: Iterable[str] | None) -> bool: ...
def get_environ_proxies(url: _Uri, no_proxy: Iterable[str] | None = ...) -> dict[Any, Any]: ...
```
It does str-specific operations on the no_proxy elements.

File: stubs/requests/requests/utils.pyi (hunk 4 above)
Comment (kkirsche) [reply stubs/requests/requests/utils.pyi]: Resolved via eb979de

File: stubs/requests/requests/utils.pyi (hunk 3 above)
Comment (kkirsche) [reply stubs/requests/requests/utils.pyi]: resolved via 4e5c15f

File: stubs/requests/requests/utils.pyi (hunk 2 above)
Comment (kkirsche) [reply stubs/requests/requests/utils.pyi]: resolved via 4e5c15f

File: stubs/requests/requests/utils.pyi (hunk 5 above)
Comment (kkirsche) [reply stubs/requests/requests/utils.pyi]: Resolved via 3c753c4

**Pull request merged**